neural_chat/pipeline/plugins/intent_detector/intent_detection.py

```diff
@@ -1,14 +1,37 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 """Function to check the intent of the input user query with LLM."""
 import transformers
 import torch
 from .prompts.prompt import generate_intent_prompt
+from neural_chat.plugins import register_plugin
+
+@register_plugin("intent_detection")
+class IntentDetector:
+    def intent_detection(self, model, tokenizer, query):

```

Comment:
why this class has no `__init__` func? it's better to have same style like other plugins